src/generator-namespace-manager.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using System.Collections.Generic;
+using ObjCRuntime;
+
+#nullable enable
+
+public class NamespaceManager {
+	public BindingTouch BindingTouch;
+	PlatformName CurrentPlatform { get { return BindingTouch.CurrentPlatform; } }
+	Frameworks? Frameworks { get { return BindingTouch.Frameworks; } }
+
+	// Where user-overrideable messaging may live
+	public string ObjCRuntime { get; private set; }
+
+	public string Messaging { get; private set; }
+
+	public ICollection<string> StandardNamespaces { get; private set; }
+	public ICollection<string> UINamespaces { get; private set; }
+	public ICollection<string> ImplicitNamespaces { get; private set; }
+	public ICollection<string> NamespacesThatConflictWithTypes { get; private set; }
+
+	public NamespaceManager (BindingTouch binding_touch, string customObjCRuntimeNS, bool skipSystemDrawing)
+	{
+		BindingTouch = binding_touch;
+
+		ObjCRuntime = String.IsNullOrEmpty (customObjCRuntimeNS)
+			? "ObjCRuntime"
+			: customObjCRuntimeNS;
+
+		Messaging = ObjCRuntime + ".Messaging";
+
+		StandardNamespaces = new HashSet<string> {
+			"Foundation",
+			"ObjCRuntime",
+			"CoreGraphics",
+		};
+
+		if (Frameworks is null)
+			// we are in a bad state
+			throw ErrorHelper.CreateError (0, "Frameworks were not loaded from platform {0}.", CurrentPlatform);
```

Comment:
New change added. @rolfbjarne  is there a needed change to get this translated, or is there no need?

Reply:
Yes, please add a new error (code) to src/Resources.resx for this.